src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStructsAndEnum.cs

```diff
@@ -1800,27 +1800,29 @@ static void Main()
             compilation.VerifyIL("S.Main",
 @"
 {
-  // Code size       57 (0x39)
+  // Code size       59 (0x3b)
   .maxstack  3
   .locals init (S V_0,
-  S V_1)
+                S V_1)
   IL_0000:  ldloca.s   V_0
   IL_0002:  initobj    ""S""
   IL_0008:  ldloca.s   V_0
   IL_000a:  ldc.i4.0
   IL_000b:  stfld      ""int S.x""
-  IL_0010:  ldloca.s   V_0
-  IL_0012:  ldloca.s   V_1
-  IL_0014:  initobj    ""S""
-  IL_001a:  ldloca.s   V_1
-  IL_001c:  ldc.i4.1
-  IL_001d:  stfld      ""int S.x""
-  IL_0022:  ldloc.1
-  IL_0023:  box        ""S""
-  IL_0028:  constrained. ""S""
-  IL_002e:  callvirt   ""bool object.Equals(object)""
-  IL_0033:  call       ""void System.Console.WriteLine(bool)""
-  IL_0038:  ret
+  IL_0010:  ldloc.0
```

Comment:
>ldloc [](start = 12, length = 5)

The only change appears to be the addition of these two instructions, which have no effect. Why did this happen?

Reply:
The receiver of `Equals` is an RValue that is lowered into passable by reference sequence. As a result we force a copy now. It is correct but a bit redundant. The JIT will understand this store/load as a NOP, so it is not a big concern. It is very specific to this particular pattern.

It can be rectified, but I did not feel like it is important to be a part of this fix.

---
In reply to: [169805013](https://github.com/dotnet/roslyn/pull/24957#discussion_r169805013) [](ancestors = 169805013)

Reply:
I've added a small peephole optimization to deal with this case. Thanks for nudging me to do it :-)

---
In reply to: [169812182](https://github.com/dotnet/roslyn/pull/24957#discussion_r169812182) [](ancestors = 169812182,169805013)